javaslang/src/main/java/javaslang/collection/TreeMap.java

```diff
@@ -477,6 +505,98 @@ public boolean isEmpty() {
         return TreeSet.ofAll(keyComparator(), iterator().map(Tuple2::_1));
     }
 
+    @Override
+    public Iterator<Tuple2<K, V>> descendingIterator() {
+        return entries.descendingIterator();
+    }
+
+    @Override
+    public SortedSet<K> descendingKeySet() {
+        return descendingMap().keySet();
+    }
+
+    @Override
+    public SortedMap<K, V> descendingMap() {
+        return new TreeMap<>(entries.descendingTree());
+    }
+
+    @Override
+    public SortedMap<K, V> subMap(Option<K> fromKey, boolean fromInclusive, Option<K> toKey, boolean toInclusive) {
+        return new TreeMap<>(
+                entries.subTree(
+                    fromKey.map(k -> Tuple.of(k, null)), fromInclusive,
+                    toKey.map(k -> Tuple.of(k, null)), toInclusive));
+    }
+
+    @Override
+    public SortedMap<K, V> subMap(K fromKey, boolean fromInclusive, K toKey, boolean toInclusive) {
```

Comment:
I think that we're creating this library because we all agree that the `JDK` way often sux :).

I would appreciate if I wouldn't have to check implementation details form methods that read like the one above (`calculate(true, 1, false)`), i.e. if nothing else, please create an
 ```java
enum Inclusion { INCLUSIVE, EXCLUSIVE }
``` and use it instead of the boolean :).